special/bin: climb up directories to find binary module

sveyret opened the pull request:
Issue #436

sveyret commented:
Note that the same problem exists for ignoreBinPackages. I am about to update it too, please wait before merging.

rumpl commented:
It would seem that you need to rebase your branch against master too

rumpl commented:
Could you add a test too ?

sveyret commented:
> It would seem that you need to rebase your branch against master too

Are you sure? It says it is already up to date…

rumpl commented:
Actually it fails because of the changes I made in a dependent build due to #433 

As soon as that build passes I'll merge it and then you can rebase.

Please don't ask what is this dependent build, I'll work on removing it :)

rumpl commented:
Some tests would be nice yes :)

Thank you for your work, it's appreciated

sveyret commented:
I added tests and also made some other modifications:

- changed the per-directory .gitignore into a single one into the test directory (they all were the same)
- removed a test which was duplicate

Each of these modification are in a separate commit for if you do not want to take them.

rumpl commented:
Thanks! I’ll take a look at it closely tomorrow but it looks good from my phone. Cheers

Also, happy halloween if you are into that kind of things :)

rumpl commented:
Unrelated but I used to work in Annecy, tres belle ville :)

sveyret commented:
@rumpl Just a reminder as you said you'd have a look at this one today… :smile:

rumpl commented on test/spec.js:
```diff
@@ -499,19 +499,6 @@ export default [
       },
     },
   },
-  {
```
Why did you remove this test?

sveyret replied:
If I understood well, it's doing exactly the same thing as 2 tests above (report unused bin dependencies if ignoreBinPackage equal false).

rumpl replied:
ooooh good catch!

rumpl commented:
Thanks for all the work you have been doing lately!

sveyret commented:
You're welcome. Actually, I'm just making it work for my project… :wink:

rumpl commented:
Need anything else or do you want me to make a release just for you?

sveyret commented:
Well, I made everything I needed, so yes, if you can make a release, that'd be great!

rumpl commented:
Lets get #451 in it too
